include/dali/core/tensor_shape.h

```diff
@@ -637,6 +638,44 @@ struct TensorListShapeBase {
     shapes.resize(num_samples * sample_dim);
   }
 
+
+  ///@{
+  /**
+   * Append this TensorListShape with shapes from other TensorListShapes.
+   */
+  void append(span<const TensorListShape<sample_ndim>> tlss) {
+    int new_samples = std::accumulate(tlss.begin(), tlss.end(), 0,
+                                      [](int acc, const auto &tls) -> int {
+                                          return acc + tls.num_samples();
+                                      });
```

Comment:
I think it's _much_ harder to read than a simple loop:
```suggestion
    int new_samples = 0;
    for (auto &tls : tlss)
      new_samples += tls.num_samples();
```

Reply:
True. I though for a minute, that `accumulate` would be vectorized under the hood, but not a chance :(